### PR TITLE
[WIP] Enable selection on FIELD_ID

### PR DESCRIPTION
**Pull request by sjperkins**
Implements #29 

- [x] Tests added / passed

  ```bash
  $ py.test --flake8 -v -s .
  ```

  If the flake8 tests fail, the quickest way to correct
  this is to run `autopep8` and then `flake8`
  to fix the remaining issues.

  ```
  $ pip install -U autopep8 flake8
  $ autopep8 -r -i .
  $ flake8 .
  ```

### CODE REVIEW

**Comment by sjperkins** [comment]
Please do not merge yet.

**Comment by sjperkins** [comment]
@o-smirnov @KshitijT Could you try this out please?

**Comment by KshitijT** [comment]
> @o-smirnov @KshitijT Could you try this out please?

With pleasure.

**Comment by KshitijT** [comment]
So.
Seems to run fine with field-ids but not with names.
This works : 
```
crystalball -sm Cyg_A_Vanilla_selfcal_ABCD-ph2-sources.txt -w 'core.reg' -sp -o FLUFFY_CAT_COLUMN -ns 1 -f '1' ../msdir/S-CYG-A-3.5-10S-4M.MS/ 

```

But not this:
```
crystalball -sm Cyg_A_Vanilla_selfcal_ABCD-ph2-sources.txt -w 'core.reg' -sp -o FLUFFY_CAT_COLUMN -ns 1 -f 'J1331+3030' ../msdir/S-CYG-A-3.5-10S-4M.MS/ 
Successful read/write open of default-locked table ../msdir/S-CYG-A-3.5-10S-4M.MS/: 33 columns, 6305000 rows
-------------------------------------------
system RAM = 503.81 GB
nr of logical CPUs = 72
nr sources = 1
nr rows    = 6305000
nr chans   = 32
nr corrs   = 4
sources per chunk = 1 (auto setting)
rows per chunk    = 13543 (auto setting)
expected memory usage = 1.86 GB
Field J1331+3030 does not exist and ignored for selection

```
here is the relevant table:
Screenshot from 2019-10-07 15-02-39

File: Crystalball/filtering.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    for field_name, field_id in list(field_names.items()):
+        exists = field_names.setdefault(str(field_id), field_id)
+
+        if not exists == field_id:
+            raise ValueError("Existing field_id mismatch "
+                             "for field %s. %d != %d"
+                             % (field_name, exists, field_id))
+
+    field_ids = []
+
```
Comment (sjperkins) [inline Crystalball/filtering.py]: @KshitijT Can you put a

```python
from pprint import pprint
pprint(field_names)
```

here and see what happens?

Comment (KshitijT) [reply Crystalball/filtering.py]: I could.

Comment (KshitijT) [reply Crystalball/filtering.py]: This is what I get:
```
crystalball -sm Cyg_A_Vanilla_selfcal_ABCD-ph2-sources.txt -w 'core.reg' -sp -o FLUFFY_CAT_COLUMN -ns 1 -f 'J1331+3030' ../msdir/S-CYG-A-3.5-10S-4M.MS/ 
Successful read/write open of default-locked table ../msdir/S-CYG-A-3.5-10S-4M.MS/: 33 columns, 6305000 rows
-------------------------------------------
system RAM = 503.81 GB
nr of logical CPUs = 72
nr sources = 1
nr rows    = 6305000
nr chans   = 32
nr corrs   = 4
sources per chunk = 1 (auto setting)
rows per chunk    = 13543 (auto setting)
expected memory usage = 1.86 GB
['J1331+3030']
{'0': 0,
 '1': 1,
 '2': 2,
 "['Cygnus-Nuc']": 0,
 "['J1331+3030']": 1,
 "['J2007+4029']": 2}
Field J1331+3030 does not exist and ignored for selection

```

**Comment by sjperkins** [comment]
And now?

**Comment by KshitijT** [comment]
Works now, awesome. Is this useful user input or what? :P

**Comment by sjperkins** [comment]
> Works now, awesome. Is this useful user input or what? :P

its amazing!

**Comment by KshitijT** [comment]
> > Works now, awesome. Is this useful user input or what? :P
> 
> its amazing!

# GoldenTeam

**Comment by sjperkins** [comment]
@paoloserra I think this can be merged, unless @KshitijT has any other objections?

**Comment by KshitijT** [comment]
> @paoloserra I think this can be merged, unless @KshitijT has any other objections?

None that I can think of right now..

**Pull request merged**